### PR TITLE
Feature/add service usage api

### PR DESCRIPTION
**Pull request by KushalD1999**
**Work Done:**
Created an API that make calls to the Service Usage API to get the enabled services and also allows to enable the services. Moreover, added an handler that checks if the required services are enabled. If not, then it allows user to enable it from the extension itself.

### CODE REVIEW

File: app/src/background/serviceUsageHandler.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import api from "debugger-extension-api";
+
+
+export async function checkRequiredServices(projectNumber) {
```
Comment (hananeikhelef) [inline app/src/background/serviceUsageHandler.ts:4]: Nit: documentation here would be cool :)

File: app/src/background/serviceUsageHandler.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        if (service['name'].match(/projects\/\d*\/services\/clouddebugger.googleapis.com/g)){
+            if (service['state'] === 'ENABLED'){
+                responseData = true
+            }
+        }
+    }
+    return responseData;
+}
+
+export async function enableRequiredService(projectNumber) {
```
Comment (hananeikhelef) [inline app/src/background/serviceUsageHandler.ts:18]: Nit: some documentation here too :)

File: api/src/core-api.js [hunk 3]
```diff
@@ -152,3 +153,42 @@ exports.deleteBreakpoint = async (debuggeeId, breakpointId) => {
...
+  );
+  const data = response.data;
+  return data;
+};
+
+/**
+ * Enables the service by making a POST call to Service Usage API .
+ * @param {String} name User's current project number.
+ */
+exports.enableServices = async (name) => {
```
[the remark this answers is not in the document]
Comment (KushalD1999) [reply api/src/core-api.js:183]: Sounds good

File: api/src/core-api.js [hunk 4]
```diff
@@ -152,3 +153,42 @@ exports.deleteBreakpoint = async (debuggeeId, breakpointId) => {
...
+  const url = "https://serviceusage.googleapis.com/v1/".concat(name, ":enable");
+  console.log(url);
```
[the remark this answers is not in the document]
Comment (KushalD1999) [reply api/src/core-api.js:185]: Removed

File: app/src/background/serviceUsageHandler.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (KushalD1999) [reply app/src/background/serviceUsageHandler.ts:4]: Sounds good. Thanks you for bringing this up

File: app/src/background/serviceUsageHandler.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+
+export async function checkRequiredServices(projectNumber) {
+    let responseData = false
+    const response =  await api.fetchServices(projectNumber);
+    const data = response.services
+    for (let service of data) {
+        if (service['name'].match(/projects\/\d*\/services\/clouddebugger.googleapis.com/g)){
+            if (service['state'] === 'ENABLED'){
+                responseData = true
```
[the remark this answers is not in the document]
Comment (KushalD1999) [reply app/src/background/serviceUsageHandler.ts:11]: Done

File: app/src/background/serviceUsageHandler.ts (hunk 1 above)
Comment (KushalD1999) [reply app/src/background/serviceUsageHandler.ts:4]: added

**Pull request merged**